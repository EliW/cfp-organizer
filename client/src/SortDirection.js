import React from 'react';

export const SortDirection = ({ sortAsc }) => {
  if (sortAsc)
    return <span className="Arrow Arrow--up"></span>; 
  else
    return <span className="Arrow Arrow--down"></span>;
};