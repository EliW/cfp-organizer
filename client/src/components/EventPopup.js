import React from 'react';

export const EventPopup = ({ event }) => (
  <React.Fragment>
    <a href={event.url} target="_blank">{event.name}</a> - { event.date } <br/>
    { event.city }, { event.country } <br/>
    <a href={event.cfpUrl} target="_blank">CFP Closes on {event.cfpClose}</a>
  </React.Fragment>
);