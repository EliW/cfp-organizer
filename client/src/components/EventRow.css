.SvgButton {
  height: 30px;
  width: 30px;
  border: none;
  background: none;
  cursor: pointer;
  margin-right: 10px;
}

.SvgButton:focus {
  outline: 3px auto #0074d9;
}

.SvgButton::-moz-focus-inner {
  border: 0;
}

.SvgButton svg {
  outline: none;
}

.u-srOnly {
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

.EventRow-button {
  border: none;
  display: block;
  font-size: 1em;
  width: 100%;
}
