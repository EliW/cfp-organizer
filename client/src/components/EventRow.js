import React from 'react';

export const EventRow = ({ className, rowClassName, event }) => (
  <div className={`${className} EventRow`}>
    <div className={rowClassName}><a href={event.url}>{event.name}</a></div>
    <div className={`${rowClassName} Table-cell--location`}>
      {event.location} 
      <span className='u-small'>{event.city}, {event.country}, {event.countryCode}, {event.continent}</span>
    </div>
    <div className={rowClassName}>{event.date}</div>
    <div className={rowClassName}><a href={event.cfpUrl}>{event.cfpClose}</a></div>
    <div className={`${rowClassName} u-small`}>{event.eventTags}</div>
  </div>
);