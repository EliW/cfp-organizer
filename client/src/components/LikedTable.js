import React from "react";
import { Table } from "./Table.js";

export const LikedTable = props => {
  const likedEventsFromStorage = localStorage.getItem(
    "cfpOrganizer-likedEvents"
  );
  let likedEvents = {};
  if (likedEventsFromStorage) {
    likedEvents = {
      ...JSON.parse(localStorage.getItem("cfpOrganizer-likedEvents"))
    };
  }

  const data = props.data.filter(event => likedEvents[event.name]);

  return <Table {...props} data={data} />;
};
