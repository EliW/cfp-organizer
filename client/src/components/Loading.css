.Loading {
  color: #666;
  margin: 6em;
  text-align: center;
}

.Loading-spinner {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.Loading-spinner div {
  position: absolute;
  width: 5px;
  height: 5px;
  background: #aaa;
  border-radius: 50%;
  animation: Loading 1.2s linear infinite;
}
.Loading-spinner .one {
  animation-delay: 0s;
  top: 29px;
  left: 53px;
}
.Loading-spinner .two {
  animation-delay: -0.1s;
  top: 18px;
  left: 50px;
}
.Loading-spinner .three {
  animation-delay: -0.2s;
  top: 9px;
  left: 41px;
}
.Loading-spinner .four {
  animation-delay: -0.3s;
  top: 6px;
  left: 29px;
}
.Loading-spinner .five {
  animation-delay: -0.4s;
  top: 9px;
  left: 18px;
}
.Loading-spinner .six {
  animation-delay: -0.5s;
  top: 18px;
  left: 9px;
}
.Loading-spinner .seven {
  animation-delay: -0.6s;
  top: 29px;
  left: 6px;
}
.Loading-spinner .eight {
  animation-delay: -0.7s;
  top: 41px;
  left: 9px;
}
.Loading-spinner .nine {
  animation-delay: -0.8s;
  top: 50px;
  left: 18px;
}
.Loading-spinner .ten {
  animation-delay: -0.9s;
  top: 53px;
  left: 29px;
}
.Loading-spinner .eleven {
  animation-delay: -1s;
  top: 50px;
  left: 41px;
}
.Loading-spinner .twelve {
  animation-delay: -1.1s;
  top: 41px;
  left: 50px;
}
@keyframes Loading {
  0%, 20%, 80%, 100% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.5);
  }
}
