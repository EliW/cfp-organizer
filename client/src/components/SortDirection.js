import React from 'react';

export const SortDirection = ({ name, sort }) => {
  if (sort.field !== name) {
    return (
      <div className="Sortable-unsorted">
        <span className="Arrow Arrow--up"></span>
        <span className="Arrow Arrow--down"></span>
      </div>
    )
  }
  else {
    if (sort.direction === 'asc')
      return <span className="Arrow Arrow--up"></span>;
    else
      return <span className="Arrow Arrow--down"></span>;
  }
};