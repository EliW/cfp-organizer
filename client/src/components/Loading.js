import React from 'react';
import './Loading.css';

export const Loading = () => (
  <React.Fragment>
    <div className="Loading">
      <div className="Loading-spinner">
        <div className="one"></div>
        <div className="two"></div>
        <div className="three"></div>
        <div className="four"></div>
        <div className="five"></div>
        <div className="six"></div>
        <div className="seven"></div>
        <div className="eight"></div>
        <div className="nine"></div>
        <div className="ten"></div>
        <div className="eleven"></div>
        <div className="twelve"></div>
      </div>
      <p>Loading...</p>
    </div>
  </React.Fragment>
);