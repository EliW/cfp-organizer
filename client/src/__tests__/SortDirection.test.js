/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import { SortDirection } from '../components/SortDirection'

let subject
let name
let sort

function loadSubject(props = {}) {
  name = 'name'
  sort = { field: 'location', direction: 'asc' } 

  subject = shallow(<SortDirection
    name={name}
    sort={sort}
    {...props}
  />)
}

describe('unsorted', () => {
  it('renders properly', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})

describe('sort asc', () => {
  it('renders properly', () => {
    loadSubject({name: 'location'})
    expect(subject).toMatchSnapshot()
  })
})

describe('sort desc', () => {
  it('renders properly', () => {
    loadSubject({ name: 'location', sort: { field: 'location', direction: 'desc' } })
    expect(subject).toMatchSnapshot()
  })
})