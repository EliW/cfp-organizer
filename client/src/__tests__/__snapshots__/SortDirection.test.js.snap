// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sort asc renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    name="location"
    sort={
        Object {
            "direction": "asc",
            "field": "location",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "Arrow Arrow--up",
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Arrow Arrow--up",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`sort desc renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    name="location"
    sort={
        Object {
            "direction": "desc",
            "field": "location",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "Arrow Arrow--down",
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Arrow Arrow--down",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`unsorted renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    name="name"
    sort={
        Object {
            "direction": "asc",
            "field": "location",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="Arrow Arrow--up"
/>,
        <span
          className="Arrow Arrow--down"
/>,
      ],
      "className": "Sortable-unsorted",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Arrow Arrow--up",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Arrow Arrow--down",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="Arrow Arrow--up"
/>,
          <span
            className="Arrow Arrow--down"
/>,
        ],
        "className": "Sortable-unsorted",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Arrow Arrow--up",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Arrow Arrow--down",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
